developing_SoCCo.py

```diff
@@ -0,0 +1,471 @@
+    percepWindowSize=3
+    riskSens=1.0
+    
+    pcE_ts,tData_ts,co2_ts = sc.iterateOneStep(pcE_ts,tData_ts, co2_ts, eff, pbc,popN,
+        percepWindowSize=3,riskSens=1.0)
+        
+    pcE_ts,tData_ts,co2_ts = sc.iterateOneStep(pcE_ts,tData_ts, co2_ts, eff, pbc,popN,
+        percepWindowSize=3,riskSens=1.0)
+        
+
+
+### TEST 3: Updating model n steps #####################################################
+
+if False: ### WORKS!
+    co2_ts=np.linspace(290, 300, 4)
+    popTotal=7130010000 # Wolfram: QuantityMagnitude[CountryData["World", "Population"]]
+    popN=sc.popIntoNgroups(popTotal,nGroups=10)
+    pcE_ts=sc.randomNormalF(5.049, 0.5, 10)
+    pcE_ts=np.atleast_2d(pcE_ts).transpose()
+    tData_ts=np.array([0,0.1,0.2,0.1]) # temperature
+    eff=sc.efficacyF(10)
+    pbc=sc.perceivedBehavioralControlF(10)
+    percepWindowSize=3
+    riskSens=1.0
+    
+    pcE_ts,tData_ts, co2_ts=sc.iterateNsteps(pcE_ts,tData_ts, co2_ts, 5, eff, pbc,popN,
+            percepWindowSize,riskSens)
+
+ 
+   
+     
+       
+         
+           
+               
+    # tData_ts.plot()
+    import matplotlib.pyplot as plt
+    hold(False)
+    
+    plt.plot(tData_ts)
+    plt.xlabel('Year')
+    plt.ylabel('T')
+    
+    plt.plot(co2_ts)
+    plt.xlabel('Year')
+    plt.ylabel('ppm')
+    
+    plt.plot(pcE_ts)
+    plt.xlabel('Year')
+    plt.ylabel('ppm')
+
+
+
+    
+    
+###### iterating function ################################################################
+
+if False:
+    
+    for i in range(10):
+        print i
+        pcE_ts,tData_ts,co2_ts = iterateOneStep(pcE_ts,tData_ts, co2_ts, eff, pbc,popN,
+            percepWindowSize=3,riskSens=1.0)
+        
+    # tData_ts.plot()
+    import matplotlib.pyplot as plt
+    hold(False)
+    
+    plt.plot(tData_ts)
+    plt.xlabel('Year')
+    plt.ylabel('T')
+    
+    plt.plot(co2_ts)
+    plt.xlabel('Year')
+    plt.ylabel('ppm')
+    
+    plt.plot(pcE_ts)
+    plt.xlabel('Year')
+    plt.ylabel('ppm')
+    
+    
+
+
+
+
+
+################ Testing of climate portion of model against observations #####
+if False:
+
+
+
+### Importing real data ################
+    
+    # CO2 data
+    co2Data=pd.read_excel("data/co2-atmospheric-mlo-monthly-scripps.xls", 
+                        sheetname='Monthly & Annual CO2 Data', skiprows=6)
+                        #http://co2now.org/Current-CO2/CO2-Now/scripps-co2-data-mauna-loa-observatory.html
+    co2Data = co2Data.set_index('Year')
+    co2Data.head()
+    co2Data.shape
+    co2A = co2Data[[u'Annual Average']] # keeping only annual means
+    #co2A = co2A.astype(np.float64)
+    co2A.shape
+    co2A.columns=['annual CO2']
+    co2A.index=pd.to_datetime(pd.Series(co2A.index), format="%Y") # unit=D) #infer_datetime_format=True) # unit=D
+    co2A.head()
+    co2A.plot()                 
+
+    # global temperature data
+    giss_temp = pd.read_table("http://data.giss.nasa.gov/gistemp/tabledata_v3/GLB.Ts+dSST.txt", sep="\s+", skiprows=7,
```

Comment:
Now that master was merged into your branch, you could use my new functions in here.